book/scale_nodes.rst

```diff
@@ -0,0 +1,355 @@
+
+   "require": {
+       ...
+       "snc/redis-bundle": "1.1.*",
+       "predis/predis": "0.8.7" #in case using Predis instead of PhpRedis
+   }
+      
+2. **execute** the "composer update"
+
+3. **register** the SncRedisBundle 
+
+  - in AppKernel.php
+
+.. code-block:: php
+
+   ...
+   public function registerBundles()
+   {
+       $bundles = array(
+       //bundles
+           new Snc\RedisBundle\SncRedisBundle()
+       );
+   ...
+
+  - or in your own bundles.yml (Acme/Bundle/AcmeBundle/Resources/config/oro/bundles.yml) 
+
+bundles:
+  ...
+  - Snc\RedisBundle\SncRedisBundle
+  ...
+
+4. **Configure** SncRedisBundle - see 
+   `the documentation <https://github.com/snc/SncRedisBundle/blob/master/Resources/doc/index.md>`_
+
+**app/config/config.yml:**
+   
+.. code-block:: yaml
+
+   snc_redis:
+       clients:  # configure phpredis client
+           default:
+               type: phpredis
+               alias: default
+               dsn: redis://localhost/1
+           doctrine:
+               type: phpredis
+               alias: doctrine
+               dsn: redis://localhost/2
+
+       doctrine: # use Redis caching for Doctrine 
+           metadata_cache:
+               client: doctrine
+               entity_manager: default
+               document_manager: default
+           result_cache:
+               client: doctrine
+               entity_manager: [default]
+           query_cache:
+               client: doctrine
+               entity_manager: default
+
+5. **Sessions storage**. By default the parameters.yml is configured to store the session in files 
+   (*session_handler: *session.handler.native_file*). The simplest way to deal with the sessions in a web farm 
+   configuration is to store them in the DB - just change the value to "*session.handler.pdo*". 
+
+   You can also store them in Redis:
+
+**app/config/config.yml:**
+   
+.. code-block:: yaml
+
+   snc_redis:
+       clients:
+           ...
+           session:
+               type: phpredis
+               alias: session
+               dsn: redis://localhost/3
+       session:  # configure sessions
+           client: session
+           prefix: session
+           use_as_default: true
+
+6. There are two abstract services used in the Oro Platform that are responsible for the cache saving.
+   *oro.file_cache.abstract*: should be used to cache the data private for each node in a web farm
+   *oro.cache.abstract*: should be used to cache the data to be shared between the nodes.
+   
+   In our case the oro.cache.abstract* is applicable.
+
+**app/config/config.yml:**   
+
+.. code-block:: yaml
+   
+   services:
+       oro.cache.abstract:
+           class: Snc\RedisBundle\Doctrine\Cache\RedisCache
+           calls:
+               - [setRedis, [@snc_redis.default]]
+
+
+7. **Annotations** cache configuration:
+
+**app/config/config.yml:**
+
+.. code-block:: yaml
+
+   framework:
+       annotations:
+           cache: oro.cache.abstract
```

Comment:
create a concrete service based on `oro.cache.abstract` and use it instead of `oro.cache.abstract`. Example of a concrete service:
```
    oro_translation.cache:
          parent: oro.cache.abstract
          calls:
              - [setNamespace, [ 'oro_trans' ] ]
```